### PR TITLE
[HttpFoundation] Make ResponseHeaderBag::makeDisposition static

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes-ish
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | #27851
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/ResponseHeaderBag.php [hunk 1]
```diff
@@ -251,51 +251,11 @@ public function clearCookie($name, $path = '/', $domain = null, $secure = false,
...
-     *                                 it can be omitted, or just copied from $filename
-     *
-     * @return string A string suitable for use as a Content-Disposition field-value
-     *
-     * @throws \InvalidArgumentException
-     *
-     * @see RFC 6266
+     * @see HeaderUtils::makeDisposition()
      */
     public function makeDisposition($disposition, $filename, $filenameFallback = '')
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/ResponseHeaderBag.php]: shall we deprecate then?

Comment (fabpot) [reply src/Symfony/Component/HttpFoundation/ResponseHeaderBag.php]: not sure it's worth it.

File: src/Symfony/Component/HttpFoundation/HeaderUtils.php [hunk 2]
```diff
@@ -143,6 +146,54 @@ public static function unquote(string $s): string
...
+     *                                 is semantically equivalent to $filename. If the filename is already ASCII,
+     *                                 it can be omitted, or just copied from $filename
+     *
+     * @return string A string suitable for use as a Content-Disposition field-value
+     *
+     * @throws \InvalidArgumentException
+     *
+     * @see RFC 6266
+     */
+    public static function makeDisposition($disposition, $filename, $filenameFallback = '')
```
Comment (OskarStark) [inline src/Symfony/Component/HttpFoundation/HeaderUtils.php]: why not using `null` as default here?

File: src/Symfony/Component/HttpFoundation/HeaderUtils.php [hunk 3]
```diff
@@ -143,6 +146,54 @@ public static function unquote(string $s): string
         return preg_replace('/\\\\(.)|"/', '$1', $s);
     }
 
+    /**
+     * Generates a HTTP Content-Disposition field-value.
+     *
+     * @param string $disposition      One of "inline" or "attachment"
+     * @param string $filename         A unicode string
+     * @param string $filenameFallback A string containing only ASCII characters that
```
Comment (OskarStark) [inline src/Symfony/Component/HttpFoundation/HeaderUtils.php]: IMO `$fallback` would be enough in this context

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/HttpFoundation/HeaderUtils.php (hunk 2 above)
Comment (fabpot) [reply src/Symfony/Component/HttpFoundation/HeaderUtils.php]: I've just moved the code and don't want to make other changes.

Comment (OskarStark) [reply src/Symfony/Component/HttpFoundation/HeaderUtils.php]: Makes sense 👍 
Thank you!

File: src/Symfony/Component/HttpFoundation/HeaderUtils.php [hunk 4]
```diff
@@ -143,6 +146,54 @@ public static function unquote(string $s): string
...
+     *
+     * @see RFC 6266
+     */
+    public static function makeDisposition($disposition, $filename, $filenameFallback = '')
+    {
+        if (!\in_array($disposition, array(self::DISPOSITION_ATTACHMENT, self::DISPOSITION_INLINE))) {
+            throw new \InvalidArgumentException(sprintf('The disposition must be either "%s" or "%s".', self::DISPOSITION_ATTACHMENT, self::DISPOSITION_INLINE));
+        }
+
+        if ('' == $filenameFallback) {
```
Comment (javiereguiluz) [inline src/Symfony/Component/HttpFoundation/HeaderUtils.php]: `===` instead of `==` ?

**Review by javiereguiluz — APPROVED** [review]
Totally unrelated to this PR, but I always find hard to understand what does "disposition attachment" and "disposition inline" mean. Normal people call this "download file" and "display file". Maybe we should add some helpers somewhere?

File: src/Symfony/Component/HttpFoundation/HeaderUtils.php (hunk 2 above)
Comment (ro0NL) [reply src/Symfony/Component/HttpFoundation/HeaderUtils.php]: IMO new code should have type declarations.

Comment (fabpot) [reply src/Symfony/Component/HttpFoundation/HeaderUtils.php]: done

**Pull request merged**